redis-utils/src/main/java/org/ballerinalang/redis/actions/ListActions.java

```diff
@@ -0,0 +1,312 @@
+     */
+    public static Object lPush(HandleValue redisDataSourceHandleValue, String redisKey, BArray values) {
+        try {
+            RedisDataSource redisDataSource = (RedisDataSource) redisDataSourceHandleValue.getValue();
+            return lPush(redisKey, redisDataSource, createStringArrayFromBArray(values)).intValue();
+        } catch (Throwable e) {
+            return BallerinaErrors.createError(REDIS_EXCEPTION_OCCURRED, e.getMessage());
+        }
+    }
+
+    /**
+     * Remove and get the first element in a list.
+     *
+     * @param redisDataSourceHandleValue redis datasource
+     * @param redisKey The key
+     * @return The value of the first element, or nil when key does not exist
+     */
+    public static Object lPop(HandleValue redisDataSourceHandleValue, String redisKey) {
+        try {
+            RedisDataSource redisDataSource = (RedisDataSource) redisDataSourceHandleValue.getValue();
+            return lPop(redisKey, redisDataSource);
+        } catch (Throwable e) {
+            return BallerinaErrors.createError(REDIS_EXCEPTION_OCCURRED, e.getMessage());
+        }
+    }
+
+    /**
+     * Get an element from a list by its index.
+     *
+     * @param redisDataSourceHandleValue redis datasource
+     * @param redisKey The key
+     * @param index The index from which the element should be retrieved
+     * @return The value at the given index
+     */
+    public static Object lIndex(HandleValue redisDataSourceHandleValue, String redisKey, int index) {
+        try {
+            RedisDataSource redisDataSource = (RedisDataSource) redisDataSourceHandleValue.getValue();
+            return lIndex(redisKey, index, redisDataSource);
+        } catch (Throwable e) {
+            return BallerinaErrors.createError(REDIS_EXCEPTION_OCCURRED, e.getMessage());
+        }
+    }
+
+    /**
+     * Insert an element before or after another element in a list.
+     *
+     * @param redisDataSourceHandleValue redis datasource
+     * @param key The key
+     * @param before boolean value representing Whether element should be inserted before or after the pivot
+     * @param pivot The pivot
+     * @param value The value
+     * @return The length of the list after the insert operation, or -1 when the value pivot not found
+     */
+    public static Object lInsert(HandleValue redisDataSourceHandleValue, String key, boolean before, String pivot,
+                                      String value) {
+        try {
+            RedisDataSource redisDataSource = (RedisDataSource) redisDataSourceHandleValue.getValue();
+            return lInsert(key, before, pivot, value, redisDataSource).intValue();
+        } catch (Throwable e) {
+            return BallerinaErrors.createError(REDIS_EXCEPTION_OCCURRED, e.getMessage());
+        }
+    }
+
+    /**
+     * Get the length of a list.
+     *
+     * @param redisDataSourceHandleValue redis datasource
+     * @param redisKey
+     * @return
```

Comment:
Missing descriptions in the param and return.

Reply:
Updated in e26d6d2